packages/social-media-app/frontend/src/identity/useIdentities.tsx

```diff
@@ -54,12 +56,48 @@ export const IdentitiesProvider = ({ children }: { children: JSX.Element }) => {
             : undefined
     );
 
+    // Create isMe function to check if a key belongs to one of the user's identities
+    const isMe = React.useCallback(
+        (identityKey: PublicSignKey | Uint8Array) => {
+            if (!peer || !devices) return false;
+
+            // Compare with the current peer's identity
+            const keyBytes =
+                identityKey instanceof PublicSignKey
+                    ? identityKey.bytes
+                    : identityKey;
+
+            // Check if the key matches our peer's key directly
+            if (
+                Buffer.from(peer.identity.publicKey.bytes).equals(
+                    Buffer.from(keyBytes)
+                )
+            ) {
+                return true;
+            }
+
+            // Check if it matches any of our connected devices
+            return devices.some((connection) => {
+                return (
+                    Buffer.from(connection.device1.publicKey).equals(
+                        Buffer.from(keyBytes)
```

Comment:
I am suprised how your editor works because Buffer.from(connection.device1.publicKey). should give Typescript compile errors. If you want to access the bytes of the key you can do

connection.device1.publicKey.bytes. 

But I am also curious what part of the chat view requires the isMe function